cleanlab/filter.py

```diff
@@ -398,6 +411,80 @@ class 0, 1, ..., K-1.
     return label_issues_mask
 
 
+def _find_label_issues_multilabel(
+    labels,
+    pred_probs,
+    return_indices_ranked_by=None,
+    rank_by_kwargs={},
+    filter_by="prune_by_noise_rate",
+    frac_noise=1.0,
+    num_to_remove_per_class=None,
+    min_examples_per_class=1,
+    confident_joint=None,
+    n_jobs=None,
+    verbose=False,
+):
+    """
+    NOT
+    """
+    num_classes = pred_probs.shape[1]
+    y_one = np.zeros((len(labels), num_classes))
+    for class_num in range(0, len(labels)):
+        for j in range(0, len(labels[class_num])):
+            y_one[class_num][j] = 1
+    if return_indices_ranked_by is None:
+        bissues = np.zeros(y_one.shape).astype(bool)
+        for class_num in range(0, num_classes):
+            pred_probabilitites = np.stack(
+                [1 - pred_probs[:, class_num], pred_probs[:, class_num]]
+            ).T
+            if confident_joint is None:
+                conf = None
+            else:
+                conf = confident_joint[class_num]
+            bissues[:, class_num] = find_label_issues(
+                y_one[:, class_num],
+                pred_probabilitites,
+                frac_noise=frac_noise,
+                rank_by_kwargs=rank_by_kwargs,
+                filter_by=filter_by,
+                multi_label=False,
+                num_to_remove_per_class=num_to_remove_per_class,
+                min_examples_per_class=min_examples_per_class,
+                confident_joint=conf,
+                n_jobs=n_jobs,
+                verbose=verbose,
+            )
+        return bissues
+    else:
+        label_issues_list = []
+
+        for class_num in range(0, num_classes):
+            pred_probabilitites = np.stack(
+                [1 - pred_probs[:, class_num], pred_probs[:, class_num]]
+            ).T
+            if confident_joint is None:
+                conf = None
+            else:
+                conf = confident_joint[class_num]
+            index_issues = find_label_issues(
+                y_one[:, class_num],
+                pred_probabilitites,
+                return_indices_ranked_by=return_indices_ranked_by,
+                rank_by_kwargs=rank_by_kwargs,
+                filter_by=filter_by,
+                multi_label=False,
+                frac_noise=frac_noise,
+                num_to_remove_per_class=num_to_remove_per_class,
+                min_examples_per_class=min_examples_per_class,
+                confident_joint=conf,
+                n_jobs=n_jobs,
+                verbose=verbose,
+            )
+            label_issues_list.append(index_issues)
```

Comment:
there is a lot of code duplication here that I think could be reduced if you only had a single one of these for loops over `class_num` but did two separate checks `if return_indices_ranked_by is None`. One before the for loop to decide what kind of object to initialize for storing for loop results, and one inside the for loop for updating that object with the output of `find_label_issues`. Eg:

``` 
if return_indices_ranked_by is None:
    bissues = np.zeros(y_one.shape).astype(bool)
else:
    label_issues_list = []

for class_num in ...
    ...
    binary_label_issues = find_label_issues(...)
    if return_indices_ranked_by is None:
        bissues[:, class_num] = binary_label_issues
    else:
        label_issues_list.append(binary_label_issues)

if return_indices_ranked_by is None:
    return bissues  # Note I'm just copying your current code here but it seems wrong, should be row-wise Or operation applied here right? 
else:
    return reduce(np.union1d, label_issues_list).astype(np.int32)
```